File: src/main/java/com/aa/tools/CalculatorController.java
package com.aa.tools;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/calculator")
public class CalculatorController {

    @Autowired
    private Calculator calculator;

    @PostMapping("/add")
    public CalculationResponse add(@RequestBody CalculationRequest request) {
        int result = calculator.add(request.getA(), request.getB());
        return new CalculationResponse(result);
    }

    @PostMapping("/subtract")
    public CalculationResponse subtract(@RequestBody CalculationRequest request) {
        int result = calculator.subtract(request.getA(), request.getB());
        return new CalculationResponse(result);
    }

    @PostMapping("/multiply")
    public MultiplyResponse multiply(@RequestBody CalculationRequest request) {
        long result = calculator.multiply(request.getA(), request.getB());
        return new MultiplyResponse(result);
    }
}

File: src/main/kotlin/com/aa/tools/CalculatorController.kt
package com.aa.tools

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/calculator")
class CalculatorController(private val calculator: Calculator) {

    @PostMapping("/add")
    fun add(@RequestBody request: CalculationRequest): CalculationResponse {
        val result = calculator.add(request.a, request.b)
        return CalculationResponse(result)
    }

    @PostMapping("/subtract")
    fun subtract(@RequestBody request: CalculationRequest): CalculationResponse {
        val result = calculator.subtract(request.a, request.b)
        return CalculationResponse(result)
    }

    @PostMapping("/multiply")
    fun multiply(@RequestBody request: CalculationRequest): MultiplyResponse {
        val result = calculator.multiply(request.a, request.b)
        return MultiplyResponse(result)
    }
}